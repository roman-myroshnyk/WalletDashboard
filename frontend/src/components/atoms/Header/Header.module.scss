@import '@/styles/colors.scss';

.header {
  position: sticky;
  top: 0;
  z-index: 1000;

  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  height: 68px !important;
  margin: 0;

  letter-spacing: -1px;

  background: white;
  background-color: $darkBlack;
}
