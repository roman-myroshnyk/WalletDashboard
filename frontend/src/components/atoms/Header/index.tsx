import { FC, ReactNode } from 'react';
import styles from './Header.module.scss';

export interface IProps {
    children: ReactNode;
}

const Header: FC<IProps> = ({ children }) => (
  <header className={styles.header}>
    {children}
  </header>
);

export default Header;
