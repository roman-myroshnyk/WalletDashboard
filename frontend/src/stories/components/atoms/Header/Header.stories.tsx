import { Meta, Story } from '@storybook/react/types-6-0';
import Header, { IProps } from '@/atoms/Header';
import Title from '@/atoms/Title';

export default {
  title: 'atoms/Header',
  component: Header,
  argTypes: {
    text: {
      controls: 'text',
    },
  },
} as Meta;

interface ITemplateProps extends IProps {
  text:string;
}
const Template:Story<ITemplateProps> = ({ ...args }) => (
  <Header {...args}>
    <Title>
      {args.text}
    </Title>
  </Header>
);

export const Default = Template.bind({});

Default.args = {
  text: 'Digital Wallet Dashboard',
};
