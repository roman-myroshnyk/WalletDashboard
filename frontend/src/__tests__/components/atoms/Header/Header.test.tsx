import { render } from '@testing-library/react';
import Header from '@/atoms/Header';

describe('atoms/Header', () => {
  it('renders Header', () => {
    const { container } = render(<Header>I am header</Header>);
    expect(container).toMatchSnapshot();
  });
});
