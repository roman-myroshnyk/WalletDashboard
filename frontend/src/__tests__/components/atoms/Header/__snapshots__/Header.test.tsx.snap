// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Header renders Header 1`] = `
<div>
  <header
    class="header"
  >
    I am header
  </header>
</div>
`;
